File: 双指针/盛最多水的容器.cpp
class Solution
{
public:
  int maxArea(vector<int> &height)
  {
    int n = height.size();
    int i = 0, j = n - 1;
    int max = 0;
    while (i < j)
    {
      int sum = min(height[i], height[j]) * (j - i);
      if (sum > max)
        max = sum;
      if (height[j] > height[i])
      {
        i++;
      }
      else
      {
        j--;
      }
    }
    return max;
  }
};

File: 贪心算法/盛最多水的容器.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let i = 0,
    j = height.length - 1;

  let res = 0;

  while (i < j) {
    res = Math.max(Math.min(height[i], height[j]) * (j - i), res);
    if (height[i] > height[j]) {
      j--;
    } else {
      i++;
    }
  }
  return res;
};
